### DISCUSSION TITLE
Reimplement support for enums in a sane way

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#6233] [Design Proposals]
The implementation of enum support in Hibernate involves a `UserType` and two `BasicValueConverter`s. This was never a very sensible way to do it, and dates back, I assume, to the time when Hibernate Annotations and Hibernate EntityManager were separate projects.

This is extremely painful to deal with in general, and in particular it makes it almost impossible to implement support for PostgreSQL `enum` types.

We need to fix this.

**Comment by gavinking** [comment]
Related: HHH-16125 , https://github.com/hibernate/hibernate-orm/pull/6232

**Comment by gavinking** [comment]
This is now fixed. @sebersole got rid of the `UserType`, and I got rid of the value converters. There's still a deprecated `EnumType` left over for certain legacy stuff (`hbm.xml`) but it's much simplified.

The real work is now handled by `EnumJavaType`, `PostgreSQLEnumJdbcType`, and `MySQLEnumJdbcType`. This  follows the usual pattern of "compositional" basic types.

**Comment by nstdio** [comment]
Sorry to post a question in closed discussion.
I wonder is there a way to explicitly specify PostgreSQL enum type? Currently implementation consider Java enum type name as PostgreSQL enum type name. Has been [comment](https://hibernate.atlassian.net/browse/HHH-16125?focusedCommentId=111108) addressed?

**Reply by gavinking** [reply]
It is this open issue: https://hibernate.atlassian.net/browse/HHH-16536
